packages/fx-core/src/plugins/resource/cicd/index.ts

```diff
@@ -0,0 +1,190 @@
+
+@Service(ResourcePluginsV2.CICDPlugin)
+export class CICDPluginV2 implements ResourcePlugin {
+  name = "fx-resource-cicd";
+  displayName = "CICD";
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  activate(projectSettings: ProjectSettings): boolean {
+    return true;
+  }
+
+  public cicdImpl: CICDImpl = new CICDImpl();
+
+  public async addCICDWorkflows(
+    context: Context,
+    inputs: Inputs,
+    envInfo: v2.EnvInfoV2
+  ): Promise<Result<any, FxError>> {
+    Logger.setLogger(context.logProvider);
+    const result = await this.runWithExceptionCatching(
+      context,
+      envInfo,
+      () => this.cicdImpl.addCICDWorkflows(context, inputs, envInfo),
+      true,
+      LifecycleFuncNames.ADD_CICD_WORKFLOWS
+    );
+
+    return result;
+  }
+
+  public async getQuestionsForUserTask(
+    ctx: Context,
+    inputs: Inputs,
+    func: Func,
+    envInfo: DeepReadonly<v2.EnvInfoV2>,
+    tokenProvider: TokenProvider
+  ): Promise<Result<QTreeNode | undefined, FxError>> {
+    const cicdWorkflowQuestions = new QTreeNode({
+      type: "group",
+    });
+
+    const whichProvider = new QTreeNode({
+      name: questionNames.Provider,
+      type: "singleSelect",
+      staticOptions: [githubOption, azdoOption, jenkinsOption],
+      title: "Select a CI/CD Provider",
+      default: githubOption.id,
+    });
+
+    const whichTemplate = new QTreeNode({
+      name: questionNames.Template,
+      type: "multiSelect",
+      staticOptions: [ciOption, cdOption, provisionOption, publishOption],
+      title: "Select template(s)",
+      default: [ciOption.id],
+    });
+
+    if (inputs.platform == Platform.VSCode) {
+      if (!inputs.projectPath) {
+        throw new NoProjectOpenedError();
+      }
+
+      const envProfilesResult = await environmentManager.listRemoteEnvConfigs(inputs.projectPath);
+      if (envProfilesResult.isErr()) {
+        throw new InternalError("Failed to list multi env.", envProfilesResult.error);
+      }
+
+      const whichEnvironment: SingleSelectQuestion = {
+        type: "singleSelect",
+        name: questionNames.Environment,
+        title: "Select an environment",
+        staticOptions: [],
+        skipSingleOption: false,
+      };
+      whichEnvironment.staticOptions = envProfilesResult.value;
+      cicdWorkflowQuestions.addChild(new QTreeNode(whichEnvironment));
+    }
+
+    cicdWorkflowQuestions.addChild(whichProvider);
+    cicdWorkflowQuestions.addChild(whichTemplate);
+
+    return ok(cicdWorkflowQuestions);
+  }
+
+  public async executeUserTask(
+    ctx: Context,
+    inputs: Inputs,
+    func: Func,
+    localSettings: Json,
+    envInfo: v2.EnvInfoV2,
+    tokenProvider: TokenProvider
+  ): Promise<Result<any, FxError>> {
+    if (func.method === "addCICDWorkflows") {
+      return await this.runWithExceptionCatching(
+        ctx,
+        envInfo,
+        () => this.addCICDWorkflows(ctx, inputs, envInfo),
```

Comment:
This method also call runWithExceptionCatching, seems it will send same telemetry event twice?

Reply:
Good catch!